doc/bootloader/README.md

```diff
@@ -0,0 +1,113 @@
+
+MCUBoot is run at boot time. In normal operation, it just jumps to the reset handler of the application firmware to run it. Once the application firmware is running, MCUBoot does not run at all.
+
+![MCUBoot boot sequence diagram](boot.png)
+
+But MCUBoot does much more than that : it can upgrade the firmware that is currently running by a new one, and it is also able to revert to the previous version of the firmware in case the new one does not run properly.
+
+To do this, it uses 2 memory 'slots' :
+- **The primary slot** (located in the internal flash memory of the nRF52832) : it contains the current firmware, the one that will be executed by MCUBoot
+- **The secondary slot** (located in the external SPI flash memory) : it is used to store the upgraded version of the firmware, when available.
+
+At boot time, MCUBoot detects that a new version of the firmware is available in the secondary slot and swaps them : the current version of the firmware is copied from the primary to the secondary slot and vice-versa.
+
+When the swapping is done, it runs the new version of the firmware from the primary slot.
+
+![MCUBoot update sequence diagram](upgrade.png)
+
+The next time MCUBoot will be run (after an MCU reset), MCUBoot will check if the new firmware ran correctly (it must set a flag in memory). If it is not the case, it'll revert to the previous version of the firmware by copying it from the secondary to the primary.
+
+![MCUBoot update sequence diagram](recover.png)
+
+You'll find more information about MCUBoot, its design and how it works in [the documentation of the project](https://docs.mcuboot.com/).
+
+The [pinetime-mcuboot-bootloader](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader) integrates MCUBoot, provides low-level drivers for the display and SPI flash memory and implements a very basic UI. See [the documentation of the project for more details about this](https://github.com/InfiniTimeOrg/pinetime-mcuboot-bootloader#boot-flow).
+
+An application firmware must implement the following functionalities to be compatible with our bootloader:
+
+- The application firmware will be flashed at **address 0x8000 of the internal flash**. It must then be **linked** with a start address of 0x8000 (instead of 0x00 by default).
+- The bootloader starts the **watchdog** just before loading the application firmware. The application firmware must **refresh the watchdog regularly**. If the watchdog is not refreshed in time, the microcontroller will reset, and the bootloader will consider the firmware as unfunctional and will revert to the previous one.
+- The application firmware must implement an **OTA procedure which copies the new firmware at offset 0x040000 of the external SPI flash memory**.
+- The application firmware must set the valid bit in flash memory (write `1` at offset 0x7BFE8 of the internal flash memory) to validate the installation. If this bit is not set, the bootloader will revert to the previous version at next reset.
```

Comment:
```suggestion
- The application firmware must set the valid bit in flash memory (write `1` at offset 0x7BFE8 of the internal flash memory) to validate the installation.
  If this bit is not set, the bootloader will revert to the previous version at next reset.
```

Reply:
Would it be better to call it "the validation bit", or perhaps "validation flag"? Saying "the valid bit" is confusing because a zero could be "the valid bit". Also using the term "bit" rather than "byte" suggests we should be more explicit about what needs to be written - would a byte equal to `0x01` be suitable, or is that byte really a bitmask of multiple flags, and they need to `OR` that location with `0x01`?

Replace version with application firmware for clarity.

Assuming a dedicated byte to contain 0x01:
```suggestion
- The application firmware must set the validation bit by **writing `0x01` at offset 0x7BFE8 of the internal flash memory** to validate the installation.
  If this bit is not set, the bootloader will revert to the previous application firmware at next reset.
```